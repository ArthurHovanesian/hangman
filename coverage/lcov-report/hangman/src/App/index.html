<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for hangman/src/App</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> hangman/src/App
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">36.21% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>21/58</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">31.25% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>5/16</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">26.09% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>6/23</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">41.18% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>21/51</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file low" data-value="App.jsx"><a href="App.jsx.html">App.jsx</a></td>
	<td data-value="35.09" class="pic low"><div class="chart"><div class="cover-fill" style="width: 35%;"></div><div class="cover-empty" style="width:65%;"></div></div></td>
	<td data-value="35.09" class="pct low">35.09%</td>
	<td data-value="57" class="abs low">20/57</td>
	<td data-value="31.25" class="pct low">31.25%</td>
	<td data-value="16" class="abs low">5/16</td>
	<td data-value="26.09" class="pct low">26.09%</td>
	<td data-value="23" class="abs low">6/23</td>
	<td data-value="40" class="pct low">40%</td>
	<td data-value="50" class="abs low">20/50</td>
	</tr>

<tr>
	<td class="file high" data-value="utilities.js"><a href="utilities.js.html">utilities.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sun Dec 02 2018 14:27:36 GMT-0800 (Pacific Standard Time)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
