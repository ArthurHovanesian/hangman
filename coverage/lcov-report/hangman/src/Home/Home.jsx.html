<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for hangman/src/Home/Home.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">hangman/src/Home</a> Home.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>6/12</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">12.5% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>1/8</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">40% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>2/5</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>6/12</span>
      </div>
    </div>
  </div>
  <div class='status-line medium'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react';
import { Link } from 'react-router-dom';
import Button from '../Button/Button.jsx';
import Title from '../Title/Title.jsx';
import styles from './Home.css';
&nbsp;
class Home extends React.Component {
  constructor() {
    super();
    this.handleChange = this.handleChange.bind(this);
    this.handleNoName = this.handleNoName.bind(this);
    this.state = {
      name: '',
      placeholder: 'NAME',
    }
  }
&nbsp;
<span class="fstat-no" title="function not covered" >  ha</span>ndleChange(name) {
    const lowerCase = <span class="cstat-no" title="statement not covered" >name.toLowerCase();</span>
<span class="cstat-no" title="statement not covered" >    if ((lowerCase.charCodeAt(name.length - 1) &gt;= 97 &amp;&amp; lowerCase.charCodeAt(name.length - 1) &lt;= 122) || (lowerCase.charCodeAt(name.length - 1) &gt;= 48 &amp;&amp; lowerCase.charCodeAt(name.length - 1) &lt;= 57)) {</span>
<span class="cstat-no" title="statement not covered" >      this.setState({name: name});</span>
    } else {
<span class="cstat-no" title="statement not covered" >      this.setState({name: name.slice(0, -1)});</span>
    };
  }
&nbsp;
<span class="fstat-no" title="function not covered" >  ha</span>ndleNoName() {
<span class="cstat-no" title="statement not covered" >    this.setState({placeholder: 'PLEASE ENTER NAME'});</span>
  }
&nbsp;
  render() {
    const { name, placeholder } = this.state;
    return (
      &lt;div className={styles.container}&gt;
        &lt;div&gt;
          &lt;Title /&gt;
        &lt;/div&gt;
        &lt;div&gt;
          &lt;input type='input' placeholder={placeholder} value={name} maxLength='10' className={styles.user} onChange={<span class="fstat-no" title="function not covered" >(e</span>vent) =&gt; <span class="cstat-no" title="statement not covered" >this.handleChange(event.target.value)}</span>&gt;&lt;/input&gt;
        &lt;/div&gt;
        {name === '' ? (
          &lt;div&gt;
            &lt;Button level={0} handleNoName={this.handleNoName}/&gt;
            &lt;Button level={1} handleNoName={this.handleNoName}/&gt;
            &lt;Button level={2} handleNoName={this.handleNoName}/&gt;
          &lt;/div&gt;
        ) : (
<span class="branch-1 cbranch-no" title="branch not covered" >          &lt;div&gt;</span>
            &lt;Link to={{pathname: `/play/${name}/easy`}}&gt;
              &lt;Button level={0}/&gt;
            &lt;/Link&gt;
            &lt;Link to={{pathname: `/play/${name}/medium`}}&gt;
              &lt;Button level={1}/&gt;
            &lt;/Link&gt;
            &lt;Link to={{pathname: `/play/${name}/hard`}}&gt;
              &lt;Button level={2}/&gt;
            &lt;/Link&gt;
          &lt;/div&gt;
        )}
&nbsp;
      &lt;/div&gt;
    )
  }
}
&nbsp;
export default Home;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sun Dec 02 2018 14:27:36 GMT-0800 (Pacific Standard Time)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
