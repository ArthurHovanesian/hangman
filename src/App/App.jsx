import React from 'react';
import axios from 'axios';
import Word from '../Word/Word.jsx';
import Keyboard from '../Keyboard/Keyboard.jsx';
import Lives from '../Lives/Lives.jsx';
import CurrentScore from '../CurrentScore/CurrentScore.jsx';
import Loading from '../Loading/Loading.jsx';
import defaultKeys from './utilities.js';
import styles from './App.css';

class App extends React.Component {
  //is props necessary?
  constructor() {
    super();
    //spinner for loading word
    this.checkUserGuess = this.checkUserGuess.bind(this);
    this.state = {
      word: [],
      notYetFound: -1,
      score: 0,
      reveal: [],
      attemptsLeft: 6,
      name: '',
      difficulty: '',
      keys: defaultKeys,
    };
  };

  componentDidMount() {
    const nameAndDifficulty = this.getNameAndDifficulty();
    this.setState({
      name: nameAndDifficulty[0],
      difficulty: nameAndDifficulty[1]
    })
    this.getWord();
  }

  componentDidUpdate(prevProps) {
    const { attemptsLeft, notYetFound, score } = this.state;
    if (attemptsLeft === 0) {
      alert('you lose')
    }
    if (notYetFound === 0) {
      this.getWord();
      this.setState({
        score: score + 1,
        notYetFound: -1,
        attemptsLeft: 6,
      })
    }
  }

  getNameAndDifficulty() {
    const path = window.location.pathname.split('/');
    const name = path[2];
    const difficulty = path[3];
    return [name, difficulty];
  }

  getWord() {
    //err handling
    //condense into one .then?
    //this is very messy
    const difficulty = this.getNameAndDifficulty()[1];
    axios.get(`/api/choose_word/${difficulty}`, {difficulty: difficulty})
      .then(res => {
        let word = res.data.toUpperCase().split('');
        let reveal = word.map(letter => ({letter: letter, show: false}));
        this.setState({
          word: word,
          notYetFound: word.length,
          reveal: reveal
        });
      })
      .catch(err => console.log(err))
      .then(() => {
        const { word } = this.state;
        let keys = Object.assign({}, defaultKeys);
        word.forEach(letter => {
          letter = letter;
          keys[letter] = true;
        })
        this.setState({keys: keys})
      })
  }

  checkUserGuess(query) {
    //there's a lot going on here
    const { word, reveal, attemptsLeft, notYetFound } = this.state;
    let numAttempts = attemptsLeft;
    let lettersFound = word.filter(letter => letter === query);
    lettersFound.length === 0 ? numAttempts -= 1 : null;
    let check = reveal.map(entry => entry.letter === query ? ({letter: entry.letter, show: true}) : entry)
    this.setState({
      notYetFound: notYetFound - lettersFound.length,
      reveal: check,
      attemptsLeft: numAttempts
    });
  }

  render() {
    console.log(this.state.word)
    const { word, reveal, keys, attemptsLeft, notYetFound, score, name, difficulty } = this.state;
    if (notYetFound > 0) {
      return (
        <div >
          <Lives attemptsLeft={attemptsLeft} />
          <CurrentScore score={score} name={name} difficulty={difficulty}/>
          <Word reveal={reveal}/>
          <Keyboard key={score} keys={keys} checkUserGuess={this.checkUserGuess} attemptsLeft={attemptsLeft}/>
        </div>
      )
    } else {
      return (
        <div>
          <Loading score={score} />
        </div>
      )
    }
  };
};

//export up at the top?
//proptypes
export default App;
