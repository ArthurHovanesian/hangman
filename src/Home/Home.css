.container {
  margin: 0 auto;
  text-align: center;
}

.user {
  width: 60%;
  text-align: center;
  font-size: 40px;
  font-family: 'Roboto', sans-serif;
  height: 50px;
  margin-top: 10%;
  margin-bottom: 10%;
  border-color: #D3D3D3;
  border-top-width: 0px;
  border-left-width: 0px;
  border-right-width: 0px;
  background-color: black;
  color: white;
  box-shadow: inset 0 -10px 10px -10px white;
}

.difficulty {
  cursor: pointer;
  width: 19%;
  height: 50px;
  font-family: 'Roboto', sans-serif;
  font-size: 40px;
  margin: 0 1% 10% 1%;
  background-color: black;
}

.start {
  cursor: pointer;
  width: 60%;
  height: 50px;
  font-family: 'Roboto', sans-serif;
  font-size: 40px;
  color: #ff0dbf;
  border-color: #ff0dbf;
  background-color: black;
}

.hovered {
  cursor: pointer;
  transform: scale(1.15);
  transition: background-color .2s ease, transform .2s ease;
}

.notHovered {
  background-color: black;
  color: #424242;
  border-color: #424242;
  transform: scale(1);
  transition: color .7s ease, box-shadow .5s ease, text-shadow .7s ease, transform .2s ease;
}
