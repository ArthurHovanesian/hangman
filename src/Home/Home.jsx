import React from 'react';
import { Link } from 'react-router-dom';
import Button from '../Button/Button.jsx';
import Title from '../Title/Title.jsx';
import styles from './Home.css';

class Home extends React.Component {
  constructor() {
    super();
    this.handleChange = this.handleChange.bind(this);
    this.handleNoName = this.handleNoName.bind(this);
    this.state = {
      name: '',
      placeholder: 'NAME',
    }
  }

  handleChange(name) {
    const lowerCase = name.toLowerCase();
    if ((lowerCase.charCodeAt(name.length - 1) >= 97 && lowerCase.charCodeAt(name.length - 1) <= 122) || (lowerCase.charCodeAt(name.length - 1) >= 48 && lowerCase.charCodeAt(name.length - 1) <= 57)) {
      this.setState({name: name});
    } else {
      this.setState({name: name.slice(0, -1)});
    };
  }

  handleNoName() {
    this.setState({placeholder: 'PLEASE ENTER NAME'});
  }

  render() {
    const { name, placeholder } = this.state;
    return (
      <div className={styles.container}>
        <div>
          <Title />
        </div>
        <div>
          <input type="input"
                 placeholder={placeholder}
                 value={name} maxLength="10"
                 className={styles.user}
                 onChange={(event) => this.handleChange(event.target.value)}
          >
          </input>
        </div>
        {name === '' ? (
          <div>
            <Button level={0}
                    handleNoName={this.handleNoName}
            />
            <Button level={1}
                    handleNoName={this.handleNoName}
            />
            <Button level={2}
                    handleNoName={this.handleNoName}
            />
          </div>
        ) : (
          <div>
            <Link to={{pathname: `/play/${name}/easy`}}>
              <Button level={0}/>
            </Link>
            <Link to={{pathname: `/play/${name}/medium`}}>
              <Button level={1}/>
            </Link>
            <Link to={{pathname: `/play/${name}/hard`}}>
              <Button level={2}/>
            </Link>
          </div>
        )}

      </div>
    )
  }
}

export default Home;
