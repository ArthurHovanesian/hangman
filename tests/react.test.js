import React from 'react';
import { shallow, mount, render } from 'enzyme';

import Home from '../src/Home/Home.jsx';
import App from '../src/App/App.jsx';

describe('<Home />', () => {
  it('renders home page', () => {
    const wrapper = shallow(<Home />);
    expect(wrapper).toMatchSnapshot();
  });
})

describe('<App />', () => {
  it('renders game', () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });
})
