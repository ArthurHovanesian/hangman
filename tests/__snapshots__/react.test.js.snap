// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders game 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Loading
        score={0}
      />,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "score": 0,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Loading
          score={0}
        />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "score": 0,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`<Home /> renders home page 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Home />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div>
          <Title />
        </div>,
        <div>
          <input
            className="user"
            maxLength="10"
            onChange={[Function]}
            placeholder="NAME"
            type="input"
            value=""
          />
        </div>,
        <div>
          <Button
            handleNoName={[Function]}
            level={0}
          />
          <Button
            handleNoName={[Function]}
            level={1}
          />
          <Button
            handleNoName={[Function]}
            level={2}
          />
        </div>,
      ],
      "className": "container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Title />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <input
            className="user"
            maxLength="10"
            onChange={[Function]}
            placeholder="NAME"
            type="input"
            value=""
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "user",
            "maxLength": "10",
            "onChange": [Function],
            "placeholder": "NAME",
            "type": "input",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Button
              handleNoName={[Function]}
              level={0}
            />,
            <Button
              handleNoName={[Function]}
              level={1}
            />,
            <Button
              handleNoName={[Function]}
              level={2}
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "handleNoName": [Function],
              "level": 0,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "handleNoName": [Function],
              "level": 1,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "handleNoName": [Function],
              "level": 2,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div>
            <Title />
          </div>,
          <div>
            <input
              className="user"
              maxLength="10"
              onChange={[Function]}
              placeholder="NAME"
              type="input"
              value=""
            />
          </div>,
          <div>
            <Button
              handleNoName={[Function]}
              level={0}
            />
            <Button
              handleNoName={[Function]}
              level={1}
            />
            <Button
              handleNoName={[Function]}
              level={2}
            />
          </div>,
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Title />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <input
              className="user"
              maxLength="10"
              onChange={[Function]}
              placeholder="NAME"
              type="input"
              value=""
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "user",
              "maxLength": "10",
              "onChange": [Function],
              "placeholder": "NAME",
              "type": "input",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Button
                handleNoName={[Function]}
                level={0}
              />,
              <Button
                handleNoName={[Function]}
                level={1}
              />,
              <Button
                handleNoName={[Function]}
                level={2}
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "handleNoName": [Function],
                "level": 0,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "handleNoName": [Function],
                "level": 1,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "handleNoName": [Function],
                "level": 2,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
